src/Cache.cpp

```diff
@@ -169,13 +169,36 @@ Cache::setup()
 
         nhlog::db()->debug("setting up cache");
 
+        // Previous location of the cache directory
+        auto oldCache = QString("%1/%2%3")
+                          .arg(QStandardPaths::writableLocation(QStandardPaths::CacheLocation))
+                          .arg(QString::fromUtf8(localUserId_.toUtf8().toHex()))
+                          .arg(QString::fromUtf8(settings->profile().toUtf8().toHex()));
+
         cacheDirectory_ = QString("%1/%2%3")
-                            .arg(QStandardPaths::writableLocation(QStandardPaths::CacheLocation))
+                            .arg(QStandardPaths::writableLocation(QStandardPaths::AppDataLocation))
                             .arg(QString::fromUtf8(localUserId_.toUtf8().toHex()))
                             .arg(QString::fromUtf8(settings->profile().toUtf8().toHex()));
 
         bool isInitial = !QFile::exists(cacheDirectory_);
 
+        // NOTE: If both cache directories exist it's better to do nothing: it
+        // could mean a previous migration failed or was interrupted.
+        bool needsMigration = isInitial && QFile::exists(oldCache);
```

Comment:
I think this should also check, that there is not already a file in the new location. Otherwise an old state may overwrite the new state on every restart.

Reply:
I don't think it will ever overwrite an existing state directory, the Qt documentation says
```
On most file systems, rename() fails only if oldName does not exist, or if a file with the new name already exists[...]
```
The semantics should be more or less the same of `mv oldName newName`

Reply:
Well, mv oldName newName with a directory will move oldName inside newName then. But I guess Qt does not do that, which should be fine in that case.